packages/expo-modules-autolinking/src/autolinking/__tests__/findModules-test.ts

```diff
@@ -430,7 +424,69 @@ describe(findModulesAsync, () => {
     });
 
     expect(result).toEqual({
-      '@expo/expo-test': expect.objectContaining({}),
+      '@expo/expo-test': expectAnyModule(),
+    });
+  });
+
+  /**
+   * TODO(@kitten): Documentation implies this should work
+   * /app
+   *   ├── /app/local-expo-module
+   */
+  it.skip('should link local non-node module (FAILURE)', async () => {
```

Comment:
This is behaviour that I think exists for `react-native-config` but for some reason doesn't seem to exist for Expo Modules? This probably will be a problem we have to reconcile when combining both algorithms (replacing `react-native-config` with `findModules`' algorithm)

I'm not sure why we allow `searchPaths` to ever change and be passed when `nativeModulesDir` is a thing? I'm also not sure if `nativeModulesDir` is always `node_modules`-structured (which I'd say is unexpected, since the name doesn't imply this) and whether it ignores nested `node_modules`.

We should probably remove `searchPaths` entirely imho

Reply:
I've found that this defaults to searching `./modules` from `projectRoot` (i.e. the app dir), which explains why it fails here. I updated the tests accordingly